pyccel/codegen/printing/ccode.py

```diff
@@ -673,8 +673,10 @@ def _print_IndexedElement(self, expr):
                     if isinstance(ind, Slice):
                         inds[i] = self._new_slice_with_processed_arguments(ind, PyccelArraySize(base, i),
                             allow_negative_indexes)
+                    else:
+                        inds[i] = Slice(ind, PyccelAdd(ind, LiteralInteger(1)), LiteralInteger(1))
```

Comment:
did you test with the decorator `allow_negative_index`

Reply:
I would also like the answer to this question which was resolved without an answer

Reply:
yes, i tested with the `allow_negative_index` decorator, it's working now.

Reply:
I looked at an example:
```
@allow_negative_index('a')
def array_view_negative_var():    
    a = np.array([[1, 2, 3], [4, 5, 6], [7, 9, 5]])
    v = -1
    x = a[v,1:]
```
is translated to:
```
    t_ndarray a;
    int64_t v;
    t_ndarray x;
    int64_t i;

    a = array_create(2, (int32_t[]){3, 3}, nd_int64);
    int64_t array_dummy_0006[] = {1, 2, 3, 4, 5, 6, 7, 9, 5};
    memcpy(a.nd_int64, array_dummy_0006, a.buffer_size);

    v = -1;
    x = array_slicing(a, 1, new_slice((v < 0) ? 3 + v : v, (v < 0) ? 3 + v : v + 1, 1), new_slice(1, a.shape[1], 1));
```
So for the negative index the new slice is:
```
new_slice(3 + v, 3 + v, 1)
```
whereas for the positive index the new slice is:
```
new_slice(v, v + 1, 1)
```
Can you explain this?

Reply:
Is it possible that this is a parenthesis issue? I.e.: `(v < 0) ? 3 + v : v + 1` is equivalent to `((v < 0) ? 3 + v : v) + 1`?

Reply:
This is not the case (see issue #648). Given that fact why does `new_slice(3 + v, 3 + v, 1)` give a  slice of size 1 and not size 0?

Reply:
`new_slice(3 + v, 3 + v, 1)` must give a slice of size 0 in this case, it should be `new_slice(3 + v, 3 + v + 1, 1)`, i think the problem is from the `_new_slice_with_processed_arguments` method which was introduced on another pull request.
@saniQQ is this a problem? using `new_slice(3 + v, 3 + v, 1)` and not `new_slice(3 + v, 3 + v + 1, 1)` for negative index ?

Reply:
It doesn't appear to be a problem as it works. But I can't seem to work out how it works. `new_slice(3 + v, 3 + v, 1)` does seem to be returning a slice of size 1 but `new_slice(v,v,1)` correctly returns size 0. I have probably missed something but I think it's worth checking what before I approve

Reply:
i think there is a bug related to the pull request that u mentioned as it should be `new_slice((v < 0) ? 3 + v : v, ((v < 0) ? 3 + v : v) + 1, 1)` there is a missing parenthesis the addition should be around the ternary
as of why they seem to have different sizes is just the fact that one is not what it's supposed to be + we are not checking for the bounds.

Reply:
Ok that makes sense as the memory is shared. I did test the shape, but probably I didn't check if we were using the ndarray shape or just the shape that pyccel calculated. If we all agree that the current code is wrong then this needs fixing before we merge. It would also be useful if the code could be tested with the fix to #648 (even if it's only locally) before merging

Reply:
i think this should be marked as resolved, I have tested using the fix of #648 and the result was :

```
void array_view_negative_var(void)
{
    t_ndarray a;
    int64_t v;
    t_ndarray x;

    a = array_create(2, (int32_t[]){3, 3}, nd_int64);
    int64_t array_dummy_0001[] = {1, 2, 3, 4, 5, 6, 7, 9, 5};
    memcpy(a.nd_int64, array_dummy_0001, a.buffer_size);

    v = -1;
    x = array_slicing(a, 1, new_slice(v < 0 ? 3 + v : v, (v < 0 ? 3 + v : v) + 1, 1), new_slice(1, a.shape[1], 1));
    free_array(a);
    free_pointer(x);
}
```